examples/systemcalls/systemcalls.c

```diff
@@ -92,7 +151,36 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
-
+	//creating a child process
+	int kidpid;
+	int status;
+	int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+	if (fd < 0) { perror("open"); abort(); }
+	
+	switch (kidpid = fork()) 
+	{
+		case -1: perror("fork"); return false;
+		
+		case 0:
+			if (dup2(fd, 1) < 0) { perror("dup2"); abort(); }
+			close(fd);
+			execv (command[0], command);
+			exit(-1);
+			
+		default:
+			close(fd);
+			if (waitpid (kidpid, &status, 0) == -1)
+				return false;
+				
+			if (WIFEXITED(status) == true)
+			{
+				if (WEXITSTATUS(status) != 0)
+					return false;
+				else 
+					return true;
+			}
+				
```

Comment:
@swapnilghonge think about how you could avoid duplicating logic between the two `do_exec` functions